torch/csrc/jit/passes/quantization.cpp

```diff
@@ -692,6 +695,149 @@ c10::optional<IValue> toTwoElementIntList(Value* v) {
   }
   return c10::nullopt;
 }
+
+script::Module findChildModule(
+    const script::Module& module,
+    const std::vector<std::string>& path) {
+  script::Module m = module;
+  for (size_t i = 0; i < path.size(); ++i) {
+    m = m.attr(path[i]).toModule();
+  }
+  return m;
+}
+
+std::string addChildModule(
+    script::Module module,
+    const script::Module& child_module,
+    const std::vector<std::string>& path) {
+  script::Module cur = module;
+  TORCH_INTERNAL_ASSERT(
+      path.size() > 0, "path must have at least one element.");
+  for (size_t i = 0; i < path.size() - 1; ++i) {
+    cur = module.attr(path[i]).toModule();
+  }
+
+  std::string cur_name = path[path.size() - 1];
+  int uid = 0;
+  std::string child_name = cur_name + "_" + c10::to_string(uid++);
+  while (module.hasattr(child_name)) {
+    child_name = cur_name + "_" + c10::to_string(uid++);
+  }
+  module.register_module(child_name, child_module.clone());
+  return child_name;
+}
+
+// Check if value is the input of the graph
+bool hitGraphInput(Value* value) {
+  Graph* graph = value->owningGraph();
+  const auto& inputs = graph->inputs();
+  return std::find(inputs.begin(), inputs.end(), value) != inputs.end();
+}
+
+Value* getModuleAccessPath(Value* instance, std::vector<std::string>& path) {
```

Comment:
This and other functions above should probably be made private (static) methods of `ModuleUseDeduper`.